### PR TITLE
Extract constraint status for public use

### PR DESCRIPTION
**Pull request by nickbabcock**
Useful for those who want to derive their own `ExceptionMapper<ConstraintViolationException>` (for instance, an exception mapper that returns JSON containing constraint violations)

### CODE REVIEW

**Comment by jplock** [comment]
Do you want to add a test around this method specifically?

**Comment by nickbabcock** [comment]
Most definitely, I'll chime back in on Monday :smile:

**Comment by nickbabcock** [comment]
Ok tests are in, a review would be great. I also added more in-depth tests for #993, which shows that messages returned because of constraint violations leave a lot wanting.

File: dropwizard-validation/src/test/java/io/dropwizard/validation/ConstraintViolationsTest.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                execValidator.validateReturnValue(
+                        new ConstraintViolationsTest(),
+                        ConstraintViolationsTest.class.getMethod("theIntTest", Integer.class),
+                        4 // the return value
+                );
+
+        assertThat(violations).hasSize(1);
+
+        Set<ConstraintViolation<?>> copy = ConstraintViolations.copyOf(violations);
+        assertThat(ConstraintViolations.determineStatus(copy)).isEqualTo(500);
```
Comment (arteam) [inline dropwizard-validation/src/test/java/io/dropwizard/validation/ConstraintViolationsTest.java]: Could we change the signature of `ConstraintViolations.determineStatus`  from
````java
public static int determineStatus(Set<ConstraintViolation<?>> violations)
````
to 
````java
public static <T> int determineStatus(Set<ConstraintViolation<T>> violations)
````

Doing so, we can avoid making a copy.

File: dropwizard-validation/src/test/java/io/dropwizard/validation/ConstraintViolationsTest.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import javax.validation.constraints.Min;
+import javax.validation.executable.ExecutableValidator;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SuppressWarnings({"UnusedDeclaration"})
+public class ConstraintViolationsTest {
+
+    private final Validator validator = Validation
```
Comment (arteam) [inline dropwizard-validation/src/test/java/io/dropwizard/validation/ConstraintViolationsTest.java]: Could we use a shorten version `Validation.buildDefaultValidatorFactory().getValidator()`?
The Hibernate provider will be loaded dynamically and we don't test validation of `Optional` values.

File: dropwizard-validation/src/test/java/io/dropwizard/validation/ConstraintViolationsTest.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        person.setName(null);
+        return person;
+    }
+
+    @Test
+    public void invalidIntReturnIs500() throws NoSuchMethodException {
+        Set<ConstraintViolation<ConstraintViolationsTest>> violations =
+                execValidator.validateReturnValue(
+                        new ConstraintViolationsTest(),
+                        ConstraintViolationsTest.class.getMethod("theIntTest", Integer.class),
```
Comment (arteam) [inline dropwizard-validation/src/test/java/io/dropwizard/validation/ConstraintViolationsTest.java]: We could add `commons-lang3` as a test dependency to the module and use static methods of `MethodUtils`.
For example, `getAccessibleMethod(getClass(), "theIntTest", Integer.class)`. Benefits - no checked exceptions.

File: dropwizard-validation/src/test/java/io/dropwizard/validation/ConstraintViolationsTest.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    public ConstraintPerson thePersonTest(@Valid ConstraintPerson person) {
+        person.setName(null);
+        return person;
+    }
+
+    @Test
+    public void invalidIntReturnIs500() throws NoSuchMethodException {
+        Set<ConstraintViolation<ConstraintViolationsTest>> violations =
+                execValidator.validateReturnValue(
+                        new ConstraintViolationsTest(),
```
Comment (arteam) [inline dropwizard-validation/src/test/java/io/dropwizard/validation/ConstraintViolationsTest.java]: Will just passing `this` work?

**Comment by arteam** [comment]
I've added some notes during breif review. Maybe @jplock will also add something.

File: dropwizard-validation/src/test/java/io/dropwizard/validation/ConstraintViolationsTest.java (hunk 1 above)
Comment (nickbabcock) [inline dropwizard-validation/src/test/java/io/dropwizard/validation/ConstraintViolationsTest.java]: Would that work? A `ConstraintViolationException` `.getConstraintViolations()` returns a `Set<ConstraintViolation<?>>`, and that would make `determineStatus` too narrow  to use. Please advise.

Implemented your other suggestions.

Comment (arteam) [inline dropwizard-validation/src/test/java/io/dropwizard/validation/ConstraintViolationsTest.java]: You are right, it's noo narrow. We need to make the generic type to extend `ConstraintViolation<?>`. Wildcard then can capture  `ConstraintViolation<?>` and `ConstraintViolation<ConstraintViolationsTest>`.
````java
public static <T extends ConstraintViolation<?>> int determineStatus(Set<T> violations)
````

**Comment by nickbabcock** [comment]
Thanks for the review, all comments were acted upon. Let me know if you find anything more.

**Comment by jplock** [comment]
I think this looks great, thanks for the contribution and for @arteam for following up!

**Pull request merged**